Remove non-standard values for `zoom`

Ahmad-S792 opened the pull request:
#### 3e0ad4d651425ffde1a2282cb561d9dead674a6d
<pre>
Remove non-standard values for `zoom`
<a href="https://bugs.webkit.org/show_bug.cgi?id=280557">https://bugs.webkit.org/show_bug.cgi?id=280557</a>
<a href="https://rdar.apple.com/136866582">rdar://136866582</a>

Reviewed by Tim Nguyen.

This patch is to align WebKit with web specification [1]:

[1] <a href="https://drafts.csswg.org/css-viewport/#zoom-property">https://drafts.csswg.org/css-viewport/#zoom-property</a>

The `zoom` property was non-standard previously but have been standardised now.
In newer standard, it does not have `document` and `reset` values.
So this patch is to remove them and Blink actually removed them in 2017.

* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/style/StyleBuilderCustom.h:
(WebCore::Style::BuilderCustom::applyValueZoom):
* LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/parsing/zoom-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/parsing/zoom-valid.html:

Canonical link: <a href="https://commits.webkit.org/284427@main">https://commits.webkit.org/284427@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d7bdf9c019c208a0662fbc8579be1653ec346338

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/69351 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/48751 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/22024 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/73433 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/20509 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/71468 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/56552 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/20360 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/55161 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/13623 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/72417 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/44480 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/59876 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/35639 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/41146 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/17306 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/18886 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/63088 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/17651 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/75145 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/13333 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/16874 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/62827 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/13372 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/59959 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/62733 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/15390 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/10749 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/4360 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/44555 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/45629 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/46824 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/45370 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/css/CSSProperties.json:
```diff
@@ -1302,18 +1302,16 @@
-            "status": "non-standard",
             "specification": {
-                "url": "https://msdn.microsoft.com/en-us/library/ms531189(v=vs.85).aspx"
+                "url": "https://drafts.csswg.org/css-viewport/#zoom-property"
```
Can you add `"category": "css-viewport"` here and define the category with the rest of the specs in this file?

Ahmad-S792 replied:
I tried adding it but it gave me style error that this category is unknown. Any reference PR, to see what else I need to do?

nt1m replied:
This is what i meant by defining the category: https://searchfox.org/wubkat/rev/b36cbce69fddb7da33823f316bd8ead5bebee970/Source/WebCore/css/CSSProperties.json#9939

Ahmad-S792 replied:
Got it. Thanks!

Ahmad-S792 replied:
Done. :-) Thanks again for your guidance.

nt1m commented on Source/WebCore/style/StyleBuilderCustom.h:
```diff
@@ -208,13 +208,6 @@ inline void BuilderCustom::applyValueZoom(BuilderState& builderState, CSSValue&
-    } else if (primitiveValue.valueID() == CSSValueReset) {
-        builderState.setUsedZoom(RenderStyle::initialZoom());
-        builderState.setZoom(RenderStyle::initialZoom());
-    } else if (primitiveValue.valueID() == CSSValueDocument) {
```
You can remove `document` from [CSSValueKeywords.in](https://searchfox.org/wubkat/source/Source/WebCore/css/CSSValueKeywords.in) because it's not used anywhere else.

Ahmad-S792 replied:
@nt1m - done. :-)

nt1m reviewed (approved):
Consider adding a test at: https://searchfox.org/wubkat/rev/b36cbce69fddb7da33823f316bd8ead5bebee970/LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/parsing/zoom-valid.html#22

and exporting it to WPT

nt1m commented on Source/WebCore/css/CSSProperties.json:
```diff
@@ -1302,18 +1302,17 @@
+                "normal"
             ],
             "codegen-properties": {
+                "initial": "initialZoom",
```
This line can probably be omitted, it's the default.

Ahmad-S792 replied:
Removed in https://github.com/WebKit/WebKit/commit/d7bdf9c019c208a0662fbc8579be1653ec346338.

Ahmad-S792 committed:
Remove non-standard values for `zoom`
https://bugs.webkit.org/show_bug.cgi?id=280557
rdar://136866582

Reviewed by Tim Nguyen.

This patch is to align WebKit with web specification [1]:

[1] https://drafts.csswg.org/css-viewport/#zoom-property

The `zoom` property was non-standard previously but have been standardised now.
In newer standard, it does not have `document` and `reset` values.
So this patch is to remove them and Blink actually removed them in 2017.

* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/style/StyleBuilderCustom.h:
(WebCore::Style::BuilderCustom::applyValueZoom):
* LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/parsing/zoom-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/parsing/zoom-valid.html:

Canonical link: https://commits.webkit.org/284427@main
